### PR TITLE
print uptime when run "hostnamectl status"

### PR DESCRIPTION
**Pull request by lucky-sideburn**

### CODE REVIEW

File: src/hostname/hostnamectl.c [hunk 1]
```diff
@@ -69,6 +73,26 @@ typedef struct StatusInfo {
...
+        int days, hours, mins;
+        char *uptime;
+        if(sysinfo(&sys_info) != 0)
+                perror("sysinfo");
+
+        days = sys_info.uptime / 86400;
+        hours = (sys_info.uptime / 3600) - (days * 24);
+        mins = (sys_info.uptime / 60) - (days * 1440) - (hours * 60);
+
+        uptime = malloc (UPTIME_BUFF_SIZE + sizeof (int) * 3);
```
Comment (utezduyar) [inline src/hostname/hostnamectl.c:88]: Missing memory check

File: src/hostname/hostnamectl.c [hunk 2]
```diff
@@ -69,6 +73,26 @@ typedef struct StatusInfo {
...
+
+        sprintf(uptime,"Uptime:  %d days, %d hours, %d minutes, %ld seconds\n",days, hours, mins, sys_info.uptime % 60);
```
Comment (utezduyar) [inline src/hostname/hostnamectl.c:90]: snprintf is safer than sprintf.

**Comment by utezduyar** [comment]
I am not sure if this enhancement will be accepted but ...
Seems like whatever hostnamectl status is presenting can also be retrieved over dbus on hostnamed.

> busctl introspect org.freedesktop.hostname1 /org/freedesktop/hostname1

**Comment by lucky-sideburn** [comment]
Hi utezduyar,

I closed the pull request because I found also another error near the malloc function.. Anyway, I am adding, in my lab server,  this feature for print the uptime but I think that is not a very useful feature :)

thanks

**Pull request closed without merging**